Files plugin tweaks

avahowell opened the pull request:
this PR makes `uploads` come after `downloads` in the file transfers list, changes styling to use the correct pointer types, and fixes a typo.

avahowell committed:
componentize file transfer, add helpful infotext

avahowell committed:
fix a few typos, styling issues

lukechampine commented on plugins/Files/js/components/filetransfers.js:
```diff
@@ -12,8 +12,8 @@ const FileTransfers = ({uploads, downloads, actions}) => {
 			</div>
-			{uploads.size > 0 ? <UploadList uploads={uploads} /> : null}
 			{downloads.size > 0 ? <DownloadList downloads={downloads} onDownloadClick={onDownloadClick} /> : null}
+			{uploads.size > 0 ? <UploadList uploads={uploads} /> : null}
```
doesn't `UploadList` need to be passed a `onUploadClick`?

avahowell replied:
no, `onUploadClick` gets passed to transfer list's `onTransferClick`, which has an es6 default parameter value.  If you pass undefined, it's the same as passing `() => {}`

avahowell commented:
dont merge yet, found a bug

avahowell committed:
move unlock prompt to buy storage, show available files in filelist

avahowell committed:
fix `ls` bug in reducer, default curried function in transferlist

also make all `available` files show up in the filelist, not just 100%
uploaded files

avahowell committed:
fix ls bug, use Path.relative instead of string manipulation

also remove a bunch of logging

avahowell committed:
fix indentation in allowance dialog

avahowell commented:
Should be ready now
